BrainPortal/app/models/message.rb

```diff
@@ -329,4 +350,30 @@ def self.parse_markup(string) #:nodoc:
     arr.join.html_safe
   end
 
+  # before_destroy callback
+  def track_resource_usage_create
+    if self.message_type != "communication"  # only user to user communication is counted so far
+      return true
+    end
+    CountResourceUsageForUserMessage.create(
+      :value              => 1,
+      :user_id            => self.sender_id,
+      :group_id           => self.group_id
```

Comment:
This doesn't do what you think. The 'group_id' attribute in a message is only used for the invitation system. It is not associated with the sender or receiver.